search_events_app/tests/views/test_views.py

```diff
@@ -167,11 +168,80 @@ def test_get_render_error(self, mock_super_get, mock_render):
 		mock_request = MagicMock()
 		presto_error = PrestoError(Exception())
 		mock_super_get.side_effect = presto_error
-		
+
 		view.get(mock_request)
 		count_calls = mock_render.call_count
 		args_calls = mock_render.call_args[0]
 
 		self.assertEqual(count_calls, 1)
 		self.assertEqual(args_calls[1], 'event_list.html')
 		self.assertEqual(args_calls[2]['error'], presto_error.message)
+
+	@patch("search_events_app.views.render")
+	def test_login_render(self, mock_render):
+		mock_request = MagicMock()
+		mock_request.method = 'GET'
+
+		views.login(mock_request)
+		count_calls = mock_render.call_count
+		args_calls = mock_render.call_args[0]
+
+		self.assertEqual(count_calls, 1)
+		self.assertEqual(args_calls[1], 'login.html')
+
+	@patch.object(DBService, 'create_connection')
+	def test_login_enter_credentials_without_error(self, mock_create_connection):
+		mock_request = MagicMock()
+		mock_request.method = 'POST'
+		mock_request.POST = {
+			'username': 'user',
+			'password': '1234'
+		}
+
+		response = views.login(mock_request)
+		response.client = Client()
+
+		self.assertRedirects(response, '/')
+
+	@patch("search_events_app.views.render")
+	@patch.object(DBService, 'create_connection')
+	def test_login_enter_credentials_with_credential_error(self, mock_create_connection, mock_render):
+		okta_error = OktaCredentialError()
+		mock_create_connection.side_effect = okta_error
+		mock_request = MagicMock()
+		mock_request.method = 'POST'
+		mock_request.POST = {
+			'username': 'user',
+			'password': '1234'
+		}
+
+
+		views.login(mock_request)
+		count_calls = mock_render.call_count
+		args_calls = mock_render.call_args[0]
+
+		self.assertEqual(count_calls, 1)
+		self.assertEqual(args_calls[1], 'login.html')
+		self.assertEqual(args_calls[2]['error'], okta_error.message)
+
+	@patch("search_events_app.views.render")
+	@patch.object(DBService, 'create_connection')
+	def test_login_enter_credentials_with_credential_error(self, mock_create_connection, mock_render):
```

Comment:
This test case has the same name as the previous one.

Reply:
Resolved